org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WEditor.java

```diff
@@ -507,9 +507,18 @@ public boolean isVisible()
         return component.isVisible();
     }
 
+    /**
+     * Indicating error with changing the style.
+     * @param error
+     */
     public void setBackground(boolean error)
     {
-
+    	if(error) {
+    		setFieldStyle("border-color: red;");
+    	}
+    	else {
+    		setFieldStyle("");
+    	}
```

Comment:
the prefer approach in zk is to use WrongValueException instead.

Reply:
Our team leading by @norbertbede tries to make iDempiere UX more attractive and follow todays standards, this ticket/improvement was also intended to this. WrongValueException renders another popup, so the user must close it before he/she can correct the input value, this means one extra step compared to the case when we indicate wrong value with red borders of the input field. WrongValueException is the preferred approach in zk, but it doesn't fulfil our vision. 

Maybe a message should be rendered inline, below input field, this could be a compromise. I add a version of it in the next commit. We think rendering multiple popups is not user friendly

![Screenshot 2022-06-22 at 14 16 36](https://user-images.githubusercontent.com/93127072/175026922-63ba56bf-cf35-4665-a42e-68d4ac4c9c70.png)
![Screenshot 2022-06-22 at 14 16 54](https://user-images.githubusercontent.com/93127072/175026927-b4179f6a-4f0a-4887-9e2b-a56623f827fd.png)
.

Reply:
> WrongValueException renders another popup, so the user must close it before he/she can correct the input value, this means one extra step compared to the case when we indicate wrong value with red borders of the input field

This is not correct, the popup that is rendered doesn't need to be closed by the user, it is automatically closed when the field that triggers the problem is changed.

You can try for example clearing the "Role" field on login and clicking the OK button, the popup shows "Fill mandatory fields: Role" - but it disappears automatically when the role is filled.